### ISSUE TITLE
CSI driver ciphers incompatible with EOS head nodes running Alma9

### ISSUE THREAD

**Issue by alexcern** [cern-eos/eosxd-csi#15]
The CSI driver enables the `arcfour-hmac-md5` cipher for compatibility with cc7 clients (https://github.com/cern-eos/eosxd-csi/blob/7d91fd8/deployments/docker/crypto-policies). This cipher is deprecated in Alma9.

On 5-Jun with https://cern.service-now.com/service-portal?id=outage&n=OTG0150286 the EOSLHCB head node was updated to Alma9. WebEOS-Alma9 clients in OKD using the CSI driver started failing to authenticate with EOSLHCB.

The reason is the selection of the deprecated cipher in the Kerberos ticket when talking to Alma9 EOS head nodes (the following was run in the eosxd-csi `automount` container):

```
# KRB5CCNAME=FILE:/var/cache/eos/fusex/credential-store/lhcb/eos-fusex-uuid-store-22ae469c-f62b-4a97-9d8f-18c036ab1c1e klist -e
Ticket cache: FILE:/var/cache/eos/fusex/credential-store/lhcb/eos-fusex-uuid-store-22ae469c-f62b-4a97-9d8f-18c036ab1c1e
Default principal: wwweos@CERN.CH

Valid starting     Expires            Service principal
06/05/24 10:38:44  06/06/24 10:38:44  krbtgt/CERN.CH@CERN.CH
        renew until 06/10/24 10:38:44, Etype (skey, tkt): aes256-cts-hmac-sha1-96, aes256-cts-hmac-sha1-96
06/05/24 12:02:29  06/06/24 10:38:44  xrootd/eoslhcb.cern.ch@CERN.CH
        renew until 06/10/24 10:38:44, Etype (skey, tkt): DEPRECATED:arcfour-hmac, aes256-cts-hmac-sha1-96
```

The head node has now been rolled back to CC7 due to this and other issues, but we had the time to verify that *removing* the deprecated `arcfour-hmac-md5` cipher from Kerberos configuration solved the issue for WebEOS-Alma9 clients. See the proposed change in https://gitlab.cern.ch/kubernetes/storage/eosxd-csi/-/merge_requests/29

However it is not confirmed if:

1. moving the deprecated cipher to the end of the list (rather than removing it altogether) would be sufficient
2. what impact the removal may have on CC7 clients (we don't have any CC7 client using eosxd-csi in OKD)

For OKD it's ok to remove the deprecated cipher altogether because we have only Alma9 clients (all Kerberos credentials management is done with the same image in both OKD PaaS and WebEOS), but perhaps it needs to be configurable for the Magnum use case if it still needs to support CC7 clients. 

EOSLHCB was the first attempt, EOS team will roll out Alma9 to EOS servers in the coming months. We should probably organize a testbed to validate the CSI driver with EOS instances running Alma9. Contacts: Cedric and Luca from <https://phonebook.cern.ch/search?q=IT-SD-PDS>

**Comment by 1602077** [comment]
Following the service meeting last week, we decided that the best approach will be to:
* drop the deprecated cipher - https://github.com/cern-eos/eosxd-csi/pull/18
* include the updated version of eosxd in the next release of cern-magnum and advise users to upgrade to latest where appropriate
* create a patch manifest to advertise the fix to users for when the cipher is deprecated and upgrading is not immediately posssible

**Comment by rochaporto** [comment]
> Following the service meeting last week, we decided that the best approach will be to:
> 
> * drop the deprecated cipher - [fix: remove deprecated arcfour-hmac-md5 cipher #18](https://github.com/cern-eos/eosxd-csi/pull/18)
> * include the updated version of eosxd in the next release of cern-magnum and advise users to upgrade to latest where appropriate
> * create a patch manifest to advertise the fix to users for when the cipher is deprecated and upgrading is not immediately posssible

We need to discuss this properly next week still.

**Comment by alexcern** [comment]
Solved with #18
